feat: update fundamental-styles to 0.7.0-rc.13

jbadan opened the pull request:
### Description

Modal
   - renamed `Dialog`
   - removed `closeProps`
   - `actions` now required
Button
   - underlying class name changes
ButtonGroup
   - classname changed to `fd-segmented-button`
- Tab
    - remove `disabled` prop
   - change underlying html - added `span` around tab text
- Shellbar
    - logo `<a>` is now `<span>`
    - added `<div>` around Identifier component
- Alert
    - renamed MessageStrip
    - classname changes 
    - removed Icon from underlying html
- Dropdown
    - renamed `Select`
    - remove `standard` prop
- ListGroup
    - renamed “List”
    - classname changes
    - so many changes! basically a new component!
- MultiInput
    - underlying html changes
    - `Popover` body now built with `List` component
    - lots of underlying html changes for validation states
    - `state` prop replaced with `validationState` - takes object
    - `localizedText` prop removed
- Combobox Input
    - underlying html changes
    - `menu` prop renamed `list` - now takes `List` component
    - `state` prop now `validationState` - takes an object 
- FormMessage
    - now a `<div>` from `<span>`
- InputGroup
    - `state` prop now `validationState` - takes an object
    - underlying html changes to include `FormMessage` when passed `validationState`
- SearchInput
    - `state` prop now `validationState` - takes an object
    - underlying html changes to include `FormMessage` when passed `validationState`https://github.com/SAP/fundamental-react/pull/885
- Toggle
    - renamed to “Switch”
    - removed size options, now use `compact` property
    - `labelProps` no longer needed, additional props will be spread to the `<label>` element
    - underlying html changes - no longer wrapped  in `FormItem`
* Checkbox
    * new `children` prop
    * `value` prop changed
    * underlying html has changed significantly 
* FormRadioItem
    * underlying html has changed significantly 
* FormMessage
    * now an internal component
* FormInput
    * removed `state` property
    * new `validationState` property
* FormItem
    * remove `isHorizontal` prop
    * remove `isInline` prop (was internal only)
* FormRadioGroup
    * now outputs `<div>` instead of Fragment
* FormLabel
    * remove `isCheckbox` prop
    * remove `isRadio` prop
    - removed `isToggle` prop


See https://github.com/SAP/fundamental-react/pull/888, https://github.com/SAP/fundamental-react/pull/885 for a divided pr

Known Issues:
Toggle/Switch
        - internalLabel prop not working in prop descriptions
        - not an accessible pattern
        - this is going to change ASAP as it is currently being redesigned 
        - icons are misaligned inside switch
Token: 
        - selected state - did not implement - what is this even used for?
Shellbar
        - `fd-shellbar__button` needs more specificity
  ListText
        - `—no-wrap` class only works in selects - this should probably be part of select, not list
Select
        - compact not working because `—fd-forms-height-compact` is not part of our postcss pipeline anymore - just SAP variables: https://github.com/SAP/fundamental-styles/pull/711
MultiInput
        - `fd-tokenizer__input` does not take priority over `fd-input` - getting the fd-input outline box (needs more specificity)
Validation states
        - validation state styling - FS has the validation message as a list item instead of a div:https://github.com/SAP/fundamental-styles/issues/712
        - “onClickOutside” is not being handled correctly

jbadan commented on .size-limit:
```diff
@@ -3,6 +3,6 @@
     webpack: true,
     path: "lib/index.js",
-    limit: "180 KB"
+    limit: "200 KB"
```
it's only 186 but I'm giving myself some headroom

skvale commented on src/Dialog/Dialog.Component.js:
```diff
@@ -0,0 +1,63 @@
+import { ComponentPage, Example } from '../_playground';
+import React, { Component } from 'react';
+
+export class DialogComponent extends Component {
```
Could this be a functional component?

jbadan replied:
oh this is copied from the previous ModalComponent - it just looks new from the name change. I don't really want to rework all the docs 😆

skvale commented on src/Dialog/Dialog.js:
```diff
@@ -1,12 +1,11 @@
 import React, { Component } from 'react';
-class Modal extends Component {
-    // select body element to add Modal component too
+class Dialog extends Component {
```
Doesn't need to happen here, but this could be a functional component pretty easy too.

skvale commented on src/List/ListIcon.test.js:
```diff
@@ -0,0 +1,26 @@
+    });
+
+    describe('Prop spreading', () => {
+        test('should allow props to be spread to the ListIcon component', () => {
```
tiny thing, this could say `List.Icon component` so it's not confused with the ListIcon thing right above

skvale commented on src/MessageStrip/__stories__/MessageStrip.stories.js:
```diff
@@ -0,0 +1,54 @@
+        <MessageStrip {...createProps()}>Default MessageStrip</MessageStrip>
+    ))
+    .add('Dismissable', () => (
+        <MessageStrip dismissable>MessageStrip</MessageStrip>
```
I think it's spelled `dismissible`.  But this does seem like the right spelling

jbadan replied:
oh good catch - I can't spell 🤣

jbadan replied:
`dismissible` is the correct spelling - I looked through the history and I've had to fix this a couple times because we(me) can't get it right. Maybe we change the prop name in the future 😆

skvale commented on src/MessageStrip/MessageStrip.test.js:
```diff
@@ -0,0 +1,95 @@
+        // expect(wrapper.state(['isActive'])).toBeTruthy();
+
+        // wrapper.find('button.fd-message-strip__close').simulate('click');
+        // expect(wrapper.state(['isActive'])).toBeFalsy();
```
?? Can we take these out now?

jbadan replied:
🔥

bcullman commented:
modal looks like it is missing some padding:
![image](https://user-images.githubusercontent.com/5056972/75597814-898db000-5a4c-11ea-9d2d-4a9fea9ae4fa.png)

TabGroup looks... not right.
![image](https://user-images.githubusercontent.com/5056972/75597792-695df100-5a4c-11ea-8c6b-cf9e017e6077.png)

validation state on multiinput cause field to shift around:
![valstates](https://user-images.githubusercontent.com/5056972/75597949-4718a300-5a4d-11ea-9382-c3d013413ce8.gif)

^ same issues in selects

popover placement in docs seems smushed:
![image](https://user-images.githubusercontent.com/5056972/75597960-69122580-5a4d-11ea-9acf-cd8b5e56fcc3.png)

jbadan commented:
@bcullman 

1. Dialog by default has no padding in the body 🤷‍♀ I added the additional sizes so that consumers can add padding as they wish. See https://fundamental-styles.netlify.com/components/dialog.html#dialog-size-modifiers
I set the defaulted to large to prevent any confusion.

1. TabGroup is updated with the new span class and I've now added the sizes now (defaulted to large like Dialog since the default from fundamental-styles has no padding). Other features see: https://github.com/SAP/fundamental-react/issues/891

1.  Popover & validation state jumping fixed on docs site - was a flexbox issue.
